### PR TITLE
rbd: use librbd.FeatureSet for features

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

go-ceph v0.3 adds constants for ImageFeature values and their names.
Instead of hardcoding "layering" in several places, use the constant
given by librbd.

The rbdVolume.ImageFeatures does not seem to be used anywhere after the
conversion. Stashing the image metadata does include the ImageFeatures
as these are retrieved when getting the image information. It is safe to
drop ImageFeatures altogether and only use the imageFeatureSet instead.

### CODE REVIEW

**Comment by nixpanic** [comment]
@phlogistonjohn some go-ceph changes for you to review :grin:

File: internal/rbd/rbd_util_test.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ package rbd
 
 import (
 	"testing"
+	"strings"
```
Comment (phlogistonjohn) [inline internal/rbd/rbd_util_test.go:21]: The linter is upset by this ordering, I think.

Comment (nixpanic) [reply internal/rbd/rbd_util_test.go:21]: Indeed, I would expect that too

**Comment by humblec** [comment]
> Nice reduction in code. :-)

Indeed, we need more :)

**Comment by Madhu-1** [comment]
@humblec PTAL

**Pull request merged**